Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1782,28 +1798,23 @@ void CGContextDrawGlyphRun(CGContextRef context, const DWRITE_GLYPH_RUN* glyphRu
 }
 #pragma endregion
 
-CGImageRef CGPNGImageCreateFromFile(NSString* path) {
-    return new CGPNGDecoderImage([path UTF8String]);
-}
-
-CGImageRef CGPNGImageCreateFromData(NSData* data) {
-    return new CGPNGDecoderImage(data);
-}
-
-CGImageRef CGJPEGImageCreateFromFile(NSString* path) {
-    return new CGJPEGDecoderImage([path UTF8String]);
-}
-
-CGImageRef CGJPEGImageCreateFromData(NSData* data) {
-    return new CGJPEGDecoderImage(data);
-}
-
 #pragma region CGBitmapContext
 struct __CGBitmapContextImpl {
     woc::unique_cf<CGImageRef> image;
 };
 
-struct __CGBitmapContext : CoreFoundation::CppBase<__CGBitmapContext, __CGBitmapContextImpl, __CGContext> {};
+struct __CGBitmapContext : CoreFoundation::CppBase<__CGBitmapContext, __CGBitmapContextImpl, __CGContext> {
+    inline ComPtr<ID2D1RenderTarget>& RenderTarget() {
+        if (__CGContext::RenderTarget() == nullptr) {
+            ComPtr<ID2D1RenderTarget> renderTarget;
+            ComPtr<ID2D1Factory> factory = _GetD2DFactoryInstance();
+            IWICBitmap* bitmap = dynamic_cast<IWICBitmap*>(_impl.image->ImageSource().Get());
+            THROW_IF_FAILED(factory->CreateWicBitmapRenderTarget(bitmap, D2D1::RenderTargetProperties(), &renderTarget));
```

Comment:
Prefer not to know about Image's internal methods; look at CGPath with its `_CGPathGetGeometry(...)`